$OpenBSD: patch-config_rules_mk,v 1.11 2012/06/11 15:33:26 landry Exp $
Add -rpath $(installdir) to be sure libxul/libxpcom can open their
depending libs when being dlopen()'ed themselves
Workarounds https://bugzilla.mozilla.org/show_bug.cgi?id=668869
--- config/rules.mk.orig	Fri Sep  7 16:04:15 2012
+++ config/rules.mk	Wed Oct 10 05:25:25 2012
@@ -521,6 +521,22 @@ endif
 endif
 endif
 
+ifeq ($(OS_ARCH),OpenBSD)
+ifdef IS_COMPONENT
+EXTRA_DSO_LDOPTS += -Wl,-Bsymbolic
+endif
+DSO_LDOPTS += -Wl,-rpath,'$(installdir)'
+OS_LDFLAGS += -Wl,-rpath,'$(installdir)'
+endif
+
+ifeq ($(OS_ARCH),Bitrig)
+ifdef IS_COMPONENT
+EXTRA_DSO_LDOPTS += -Wl,-Bsymbolic
+endif
+DSO_LDOPTS += -Wl,-rpath,'$(installdir)'
+OS_LDFLAGS += -Wl,-rpath,'$(installdir)'
+endif
+
 #
 # HP-UXBeOS specific section: for COMPONENTS only, add -Bsymbolic flag
 # which uses internal symbols first
